### PR TITLE
bug(gemfury): Adds tag validation to reject tags not adhering to pip Version parsing

### PR DESCRIPTION
**Pull request by egalpin**

### CODE REVIEW

**Review by eddiedialpad — APPROVED** [review]
Approve.
• prevent tagging versions for gemfury not conforming to semvar or dev suffix

**Comment by egalpin** [comment]
@TheKevJames @eddiedialpad thoughts on whether I should release this as a major/minor/patch version? I would tend to think major since it changes behaviour somewhat significantly, but it's kind of fixing a bug in my opinion i.e. it should have behaved this way previously. Thoughts? Implications?

**Review by harsh-dialpad — APPROVED** [review]
LGTM

**Comment by eddiedialpad** [comment]
TLDR: **Major** -- updaters must read commit notes to know if updating this req will break their code.

Does it count as a _major_ change, if it's just explicitly invalidating things that were supposed to be invalid by convention anyway?

... Hmm ... existing uses ... possibly now invalid. Even though those then-valid cases are more like unspecified behaviour ...

Interesting: I vote this is either Major (could but shouldn't break existing functionality) or Patch (reduces number of unspecified cases).

Conservatively: **Major.** So if someone is updating, they'll have to read the commit notes and decide if they need to do anything for compatibility for their use, before upgrading reqs.

**Comment by egalpin** [comment]
Ya, I'm mostly inclined to go major as well. Alright, I'll do that and then backpedal as required. Nothing stopping us from doing both if we Really Need To™
